src/kernel-install/90-loaderentry.install.in

```diff
@@ -115,6 +115,43 @@ install -m 0644 "$KERNEL_IMAGE" "$KERNEL_DEST" || {
 }
 chown root:root "$KERNEL_DEST" || :
 
+if [ -n "$KERNEL_INSTALL_CONF_ROOT" ]; then
+    if [ -f "$KERNEL_INSTALL_CONF_ROOT/devicetree" ]; then
+        read -r DEVICETREE <"$KERNEL_INSTALL_CONF_ROOT/devicetree"
```

Comment:
Not familiar with device tree, but the file always contains single line?

Reply:
No the device tree blobs are a binary format. This file contains the filename of the device tree blob to use. Eg. `broadcom/bcm2711-rpi-4-b.dtb`.

On debian this file would then be found at `/usr/lib/linux-image-$KERNEL_VERSION/broadcom/bcm2711-rpi-4-b.dtb` and copied to $ENTRY_DIR_ABS like the kernel and initrd.

Reply:
I meant about `"$KERNEL_INSTALL_CONF_ROOT/devicetree"` or so. The file contains only single line? Otherwise, the whole searching logic below will not work.

Reply:
Yes, that's what I'm try to say. `"$KERNEL_INSTALL_CONF_ROOT/devicetree"` just contains a single line with the path, eg. `broadcom/bcm2711-rpi-4-b.dtb`.

Reply:
I see. Thanks.